src/components/TransactionPreview/ActionPreview.tsx

```diff
@@ -432,14 +459,27 @@ const ActionPreview = ({
   editButtonDisabled = false,
   showSignButton = false,
   showEditButton = false,
+  isSubmitted = false,
   setIsTransactionDone,
   showStatus = true,
   showGasAssetSelect = false,
 }: TransactionPreviewInterface) => {
-	const { accountAddress, providerAddress } = useEtherspot();
-	const theme: Theme = useTheme();
+  const [timer, setTimer] = useState(0);
+  const { accountAddress, providerAddress } = useEtherspot();
+
+  const theme: Theme = useTheme();
+
+  const { preview, chainId, type, estimated, isEstimating } = crossChainAction;
+
+  useEffect(() => {
+    if (isSubmitted) {
+      let interval = setInterval(() => {
+        setTimer(timer + 1);
+      }, 1000);
+      return () => clearInterval(interval);
+    }
```

Comment:
make sure u always return timeout clear, move this line outside `if` and just add simple condition if interval is set before clearing one